test/test_torch.py

```diff
@@ -6686,6 +6686,30 @@ def test_masked_fill(self):
         dst2.masked_fill_(dst2 > 0, val)
         self.assertEqual(dst, dst2, 0)
 
+    def test_one_hot(self):
```

Comment:
Can you add a test case for the following two inputs:
1) scalar tensor `torch.one_hot(torch.tensor(1))`
2) tensor with 0-size dim `torch.one_hot(torch.empty([2, 0], dtype=torch.long)`

Reply:
(To clarify, these don't have to work, if they don't work we should print out a reasonable error message for them)